server/src/main/java/org/elasticsearch/index/mapper/DynamicFieldsBuilder.java

```diff
@@ -293,7 +293,7 @@ public void newDynamicDoubleField(ParseContext context, String name) throws IOEx
 
         @Override
         public void newDynamicBooleanField(ParseContext context, String name) throws IOException {
-            createDynamicField(new BooleanFieldMapper.Builder(name), context);
+            createDynamicField(new BooleanFieldMapper.Builder(name, null), context);
```

Comment:
I wonder if instead of null we should have some impl of ScriptCompiler that throws UnsupportedOperationException when used.

Reply:
++, either here or in a follow up - it will be useful on the other field mappers as well